event/event.go

```diff
@@ -256,7 +256,13 @@ func (r *MessageSubscription) Serve(handler MessageHandler) error {
 					const size = 64 << 10
 					buf := make([]byte, size)
 					buf = buf[:runtime.Stack(buf, false)]
-					slog.Error("panic: message subscription: handling message", "message", msg, "stack_trace", string(buf))
+					messageID, publishTime := getMetadata(msg)
+					slog.Error("panic: message subscription: handling message",
+						"error", err,
+						"message_body", string(msg.Body),
+						"message_id", messageID,
+						"publish_time", publishTime,
```

Comment:
I got a bit confused about what you meant by message metadata... what `getMetadata` returns is completely different from what `msg.Metadata` contains, right?

Reply:
not completely different, I think it is more like a subset, I need to remember what is this function used for :laughing: (this is confusing because some brokers talk about metadata and attributes, in some of them the concepts are inverted...so abstracting this is hard..but there is usually both things).

Reply:
Ah yeah looked at the function, yes that is a more low level thing that is used to get the metadata from the underlying google thing. It is used here so you don't need to call it again: https://github.com/birdie-ai/golibs/blob/5cb693a7a0a035e413dba3e15ab4bdc0c3220f00/event/event.go#L251

But I also got confused.... we have two messages, the google one and the one we actually handle: https://github.com/birdie-ai/golibs/blob/5cb693a7a0a035e413dba3e15ab4bdc0c3220f00/event/event.go#L272

I thought "msg" would be ours, but it is the google one. My previous code was even wronger than I imagined, sorry I built this error handling and never tested it...should have done some manual testing at least x_x. Now I get why the error message was extra odd....I was printing the library type...not our message type (that is much simpler).

Reply:
What we could do here is before the recover create our own msg, our type, and then use that on the logging for the recover. Maybe call the other one "rawMsg" and then ours "msg".

Reply:
Sorry Im giving you so much work, I did this in a hurry and now you are paying the price for my incompetence :disappointed: 

![](https://github.com/katcipis/memes/blob/master/shame.gif?raw=true)

Reply:
Done